### PR TITLE
RD-6387 Add documentation for Secret Providers widget

### PR DESCRIPTION
**Pull request by PatrycjaKubiczek**
ticket: RD-6387
https://docs.cloudify.co/staging/RD-6387/working_with/console/widgets/secretproviders/

PR in Stage: https://github.com/cloudify-cosmo/cloudify-stage/pull/2408

### CODE REVIEW

File: content/working_with/console/widgets/secretProviders.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+title: Secret Providers Management
+category: Widgets
+draft: false
+---
+
+Displays all the secret providers visible to the logged-in user in the current tenant (including global secrets).
```
Comment (qooban) [inline content/working_with/console/widgets/secretProviders.md:7]: What does "(including global secrets)" mean? 🤔

File: content/working_with/console/widgets/secretProviders.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+![secret-providers]( /images/ui/widgets/secret-providers.png )
+
+The widget provides the following information:
+
+* **Provider name** - The name of the secret provider.
+* **Provider type** - The type of the secret provider.
+* **Creation time**
+* **Last update time**
+
+The right column of the table allows permitted users to edit the secret provider’s values or delete it.
```
Comment (qooban) [inline content/working_with/console/widgets/secretProviders.md:18]: I would say:
```suggestion
The right column of the table allows permitted users to edit the secret provider’s details or delete it.
```

File: content/working_with/console/widgets/secretProviders.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+To better understand how secret providers work in {{< param product_name >}}, go to [Secret Store page]({{< relref "developer/blueprints/spec-secretstore.md" >}}) or [Using the Secret Store page]({{< relref "working_with/manager/using-secrets.md" >}}).
```
Comment (qooban) [inline content/working_with/console/widgets/secretProviders.md:20]: At https://docs.cloudify.co/staging/dev/developer/blueprints/spec-secretstore/ I don't see any specific description about secret providers.
If needed we can refer to https://docs.cloudify.co/staging/dev/cli/orch_cli/secrets-providers/.

BTW I saw that at https://docs.cloudify.co/staging/dev/working_with/manager/using-secrets/ there are Secrets widget screenshots to update.

File: content/working_with/console/widgets/secrets.md [hunk 4]
```diff
@@ -17,7 +17,8 @@ If the secret’s value is hidden and the logged-in user isn’t the secret’s
 ![hidden-value-user]( /images/ui/widgets/hidden_secret_unauth_user.png )
```
Comment (qooban) [inline content/working_with/console/widgets/secrets.md:17]: Probably worth to update two Secrets widget screenshots used in this page to be up to date.

File: content/working_with/console/widgets/secrets.md [hunk 5]
```diff
@@ -17,7 +17,8 @@ If the secret’s value is hidden and the logged-in user isn’t the secret’s
...
 
 
-* **Hidden Value** Indicates if the secret’s value is hidden of not. If the logged-in user is the secret’s creator, or has admin/manager permissions in the tenant, checking/unchecking this field will be enabled, and will make the secret hidden/non-hidden.
+* **Hidden** Indicates if the secret’s value is hidden of not. If the logged-in user is the secret’s creator, or has admin/manager permissions in the tenant, checking/unchecking this field will be enabled, and will make the secret hidden/non-hidden.
```
Comment (qooban) [inline content/working_with/console/widgets/secrets.md:20]:
```suggestion
* **Hidden** Indicates if the secret’s value is hidden or not. If the logged-in user is the secret’s creator, or has admin/manager permissions in the tenant, checking/unchecking this field will be enabled, and will make the secret hidden/non-hidden.
```

File: content/working_with/console/widgets/secretProviders.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+---
+
+Displays all the secret providers visible to the logged-in user in the current tenant (including global secrets). 
+
+![secret-providers]( /images/ui/widgets/secret-providers.png )
+
+The widget provides the following information:
+
+* **Provider name** - The name of the secret provider.
+* **Provider type** - The type of the secret provider.
```
Comment (qooban) [inline content/working_with/console/widgets/secretProviders.md:14]: Optional:

Probably we can have just:
```suggestion
* **Provider name**
* **Provider type**
```
without losing much details.

File: content/working_with/console/widgets/secretProviders.md (hunk 1 above)
Comment (PatrycjaKubiczek) [reply content/working_with/console/widgets/secretProviders.md:7]: I based my documentation mainly on secrets widget https://docs.cloudify.co/latest/working_with/console/widgets/secrets/ but I guess I copied too much this time 😅 adjusted: https://github.com/cloudify-cosmo/docs.getcloudify.org/pull/2498/commits/18bfea37a46cfa875ff053049ef8580e7be1a7fe

File: content/working_with/console/widgets/secretProviders.md (hunk 3 above)
Comment (PatrycjaKubiczek) [reply content/working_with/console/widgets/secretProviders.md:20]: do you think I should also create https://docs.cloudify.co/staging/dev/working_with/manager/using-secrets/ & https://docs.cloudify.co/staging/dev/developer/blueprints/spec-secretstore/ but for secret providers store? Not sure if the scope of this ticket is only specific for widget documentation or wider? 🤔

Comment (qooban) [reply content/working_with/console/widgets/secretProviders.md:20]: No, no, I don't think so. It's not on our side to create these kind of pages.

Comment (PatrycjaKubiczek) [reply content/working_with/console/widgets/secretProviders.md]: okay, so I removed that line - https://github.com/cloudify-cosmo/docs.getcloudify.org/pull/2498/commits/b4a7b7cc6291f071c85a58ff66fe2b1d7b6d0fe3 ✔️

File: content/working_with/console/widgets/secrets.md (hunk 4 above)
Comment (PatrycjaKubiczek) [reply content/working_with/console/widgets/secrets.md:17]: I noticed that there's separate ticket about updating Secrets documentation: https://cloudifysource.atlassian.net/browse/RD-6402 do you think I should update images as a part of this PR?

File: content/working_with/console/widgets/secretProviders.md (hunk 1 above)
Comment (qooban) [reply content/working_with/console/widgets/secretProviders.md:7]: So are we going to keep "(including global secrets)"?

File: content/working_with/console/widgets/secrets.md (hunk 4 above)
Comment (qooban) [reply content/working_with/console/widgets/secrets.md:17]: Right, let's keep it for RD-6402.

**Review by qooban — COMMENTED** [review]
Added one comment.

File: content/working_with/console/widgets/secretProviders.md (hunk 1 above)
Comment (PatrycjaKubiczek) [reply content/working_with/console/widgets/secretProviders.md]: https://github.com/cloudify-cosmo/docs.getcloudify.org/pull/2498/commits/b908e20bef3fe9922f2ad05efc655306ab170e33 no, no, removed ✔️

**Review by qooban — APPROVED** [review]

**Pull request merged**